src/fsharp/CompileOps.fs

```diff
@@ -1835,6 +1835,67 @@ let ComputeMakePathAbsolute implicitIncludeDir (path: string) =
     with 
         :? System.ArgumentException -> path  
 
+//
+// Add your features to this List - in code use languageVersion.SupportsFeature(LanguageFeatures.yourFeature) 
+// a return value of false means your feature is not supported by the users language selection
+// All new language features added from now on must be protected by this.
+// Note:
+//   *  The fslang design process will require a decision about feature name and whether it is required.
+//   *  When a feature is assigned a release language, we will scrub the code of feature references and apply
+//      the Release Language version.
+//
+[<RequireQualifiedAccess>]
+type LanguageFeature =
+    | LanguageVersion47 = 0
+    | Nullness = 1000
+    | ScriptingPackageManagement = 1001
+
+type LanguageVersion (specifiedVersion) =
+    static let previewVersion = 4.8m                // Language version when preview specified
+    static let defaultVersion = 4.7m                // Language version when default specified
+    static let latestVersion = defaultVersion       // Language version when latest specified
+    static let latestMajorVersion = 4.7m            // Language version when latestmajor specified
+
+    static let validOptions = [| "preview"; "default"; "latest"; "latestmajor" |]
+    static let languageVersions = set [| latestVersion |]
+
+    static let features = dict [|
+        LanguageFeature.LanguageVersion47, latestVersion
+        LanguageFeature.Nullness, previewVersion
+        LanguageFeature.ScriptingPackageManagement, previewVersion
+        |]
+
+    static let dumpAllowedValues () =
```

Comment:
Nit: this should just be a member

Reply:
It's a private helper function.

Reply:
Kevin is right - no need to make it a member